### PR TITLE
DOC: Specify custom callable option for metric in cluster.hierarchy.fclusterdata

### PR DESCRIPTION
**Pull request by adeak**
The [documentation of `cluster.hierarchy.fclusterdata`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.hierarchy.fclusterdata.html) suggested that only `str` can be passed as `metric`. However, the implementation [only passes the metric to `spatial.distance.pdist`](https://github.com/scipy/scipy/blob/master/scipy/cluster/hierarchy.py#L2693) which readily accepts custom callables. Another close moving part, [`cluster.hierarchy.linkage`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.hierarchy.linkage.html#scipy.cluster.hierarchy.linkage), also accepts custom callables for metric. Calling `fclusterdata` with a custom callable seems to work fine, as expected from the implementation.

The bottom line is that this seems like a simple doc bug to me, so I added the option for functions to be passed as the `metric` keyword in the documentation of `fclusterdata`.

### CODE REVIEW

**Comment by adeak** [comment]
The change is doc-only and trivial so I'm sure the failing test is unrelated. Since I see a bunch of other PRs with failing tests I won't trigger another round of tests for the time being.

**Comment by rgommers** [comment]
Merged, thanks @adeak. This change is clearly correct, so in it goes. There were no existing tests for `fclusterdata(..., metric=...)` so I wouldn't want to require them here. If you'd want to add some for both string and callable metrics, that would be welcome though.
